sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/FunctionTableSubqueryArgumentExpression.scala

```diff
@@ -165,20 +165,18 @@ case class FunctionTableSubqueryArgumentExpression(
    * when the partition has changed.
    */
   lazy val partitioningExpressionIndexes: Seq[Int] = {
-    val extraPartitionByExpressionsToIndexes: Map[Expression, Int] =
-      extraProjectedPartitioningExpressions.map(_.child).zipWithIndex.toMap
-    partitionByExpressions.map { e =>
-      subqueryOutputs.getOrElse(e, extraPartitionByExpressionsToIndexes(e) + plan.output.length)
+    (0 until extraProjectedPartitioningExpressions.length).map { i =>
+      if (selectedInputExpressions.nonEmpty) {
+        selectedInputExpressions.length + i
```

Comment:
Why is this related to `selectedInputExpressions`?

Reply:
This will break
https://github.com/apache/spark/blob/51188ed56620c17841dac60f7f2a9330a1087d6e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/FunctionTableSubqueryArgumentExpression.scala#L124
if `selectedInputExpressions.length` and `plan.output.length` are different.

Reply:
Good catch. This is related to `selectedInputExpressions` because if they are set, we want to use different indexes here [1]. I switched to update that location instead, which should no longer break the location you mentioned.

[1] https://github.com/apache/spark/blob/201df0d7ac81f6bd5c39f513b0a06cb659dc9a3f/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L2211